### DISCUSSION TITLE
QtWebEngine can't parse pdf extension manifest

### DISCUSSION THREAD

**Discussion by crides** [qutebrowser/qutebrowser#7536] [Q&A] [answered]
Hi, the other day I deleted `~/.local/share/QtWebEngine` to save space (cuz I thought qute uses its own directory for that stuff), and now qute can't start. I tried copying over qute's version (from `~/.local/share/qutebrowser/webengine`) but it's still the same. stdout/stderr has:

```
[11597:11597:1227/194323.707903:ERROR:extension_system_qt.cpp(121)] Failed to parse extension manifest.
```

And then dumps core. From the backtrace it seems like Qt has trouble parsing the pdf manifest. Any ideas how I can resolve this? Thanks!

System: `Linux Aostro-5468 6.1.1-arch1-1 #1 SMP PREEMPT_DYNAMIC Wed, 21 Dec 2022 22:27:55 +0000 x86_64 GNU/Linux`, recently updated

`python-pyqt5`: `5.15.7-3`

`python-pyqt5-webengine`: `5.15.6`

qute: a custom fork, for tree tabs and some other changes, merged to `f77ccd022` on upstream

**Comment by crides** [comment, marked as answer]
**Edit:** Answer in thread, not in this reply

Okay I have managed to solve that issue by uninstall and reinstalling various qt and webengine packages, and fiddling with versions (the fact that they don't have the same patch version is weird). It seems that the arch `python-pyqt5-*` packages and the corresponding packages in PyPI have weird interactions.

However, now the `QtWebEngineProcess`'es are crashing immediately upon starting qute (exit status 159). Analysing the core with gdb shows that it always crashes with `SIGSYS bad system call` on `clone3`. Using `--qt-flag single-process` somewhat works (for example trying to log in to Google causes a `SIGTRACE` for some reason).

**Reply by toofar** [reply]
What version of webengine do you have now? If it's an old one you might be running into https://github.com/qutebrowser/qutebrowser/issues/7353. Try starting with `-s qt.chromium.sandboxing disable-seccomp-bpf`.

**Reply by crides** [reply]
the Arch package: `qt5-webengine: 5.15.7-2`

the PyPI packages: 

```
PyQtWebEngine                 5.15.5
PyQtWebEngine-Qt5             5.15.2
```

**Reply by crides** [reply]
and the sandboxing option does work

**Reply by The-Compiler** [reply]
It sounds like you still are running a Qt 5 from PyPI, which is too old for Archlinux' glibc. I'd heavily suggest uninstalling that and relying on the Archlinux packages instead. In general, using [virtual environments](https://chriswarrick.com/blog/2018/09/04/python-virtual-environments/) (or [pipx](https://pypa.github.io/pipx/) for applications) helps with those kind of issues, because manually installed Python packages are separate from the system-installed ones.

**Reply by crides** [reply]
Okay apparently there were a few more packages I need to reinstall to make that work. Now that seems to work properly, other than that cookies don't seem to be preserved between qute launches (I need to relogin to github, google & discord doesn't remember my accounts etc.). There doesn't seem to be anything related in stdout or `.xsession-errors`. Any ideas about this one?

**Reply by toofar** [reply]
Sometimes (all the time?) when you downgrade the older webengine/chromium stuff doesn't understand the format of the files that the newer version wrote. You might have to nuke (or move out of the way) your cookies file ([for example](https://github.com/qutebrowser/qutebrowser/issues/5847#issuecomment-718962766)). I'm not sure if this situation applies to other stuff in the webengine folder like local storage.

**Reply by crides** [reply]
everything under `~/.local/share/qutebrowser/webengine` has been gone, but it's still not working (there's no new files created). Also my Qt packages should be up to date with Arch.

**Reply by toofar** [reply]
:confused:
Can you post the output of `:version` please? Not creating new stuff is strange, maybe it's pointing elsewhere? (In the Paths section).

**Reply by crides** [reply]
```
Paths:
cache: $HOME/.cache/qutebrowser
config: $HOME/.config/qutebrowser
data: $HOME/.local/share/qutebrowser
runtime: /run/user/1000/qutebrowser
system data: /usr/share/qutebrowser
```

Also above this (probably doesn't matter?):
```
Imported from $HOME/gitproj/qutebrowser/qutebrowser
Using Python from /usr/bin/python3
Qt library executable path: /usr/lib/qt/libexec, data path: /usr/share/qt
```

**Reply by toofar** [reply]
That all looks right. Very strange. I'm all out of ideas sorry.

**Reply by crides** [reply]
I... changed qute in the webengines profile stuff, when I was debugging the previous issues. Now it's fine. Thanks for the help!

**Reply by dfaure-kdab** [reply]
`SIGSYS bad system call on clone3` was https://bugreports.qt.io/browse/QTBUG-96214 which got fixed in Qt 5.15.8
